### DISCUSSION TITLE
Comments with go:generate

### DISCUSSION THREAD

**Discussion by mactsouk** [golang/vscode-go#1537] [Help]
Hello.

Is there a way to disable automatic formatting for comments with `go:generate`?

Thanks!

**Comment by stamblerre** [comment]
@mactsouk: Can you please give an example/repro case for the issue you're encountering?

**Comment by mactsouk** [comment]
If `go:generate` comments are in `main()` function, then the following

```
func main() {
	fmt.Println("Hello there!")
//go:generate echo Hello!
}
```

is auto-formatted to the next:

```
func main() {
	fmt.Println("Hello there!")
	//go:generate echo Hello!
}
```

which does not work with the `go generate` command.

**Reply by stamblerre** [reply]
I think it is fairly uncommon to see //go:generate comments in function bodies rather than at the top-level, which is probably why that formatting happens. Still, VS Code Go just uses the default `gofmt` formatting, so if you have an issue with it, I suggest filing an issue with gofmt at https://golang.org/issue/new.

**Reply by mactsouk** [reply]
@stamblerre You might be correct about not having `//go:generate` comments inside `main()`.
Thanks for answering. 😄
